powerauth-restful-security-spring/src/main/java/io/getlime/security/powerauth/rest/api/spring/annotation/PowerAuthAnnotationInterceptor.java

```diff
@@ -96,14 +120,103 @@ public boolean preHandle(HttpServletRequest request, HttpServletResponse respons
                     );
                     request.setAttribute(PowerAuth.AUTHENTICATION_OBJECT, authentication);
                 } catch (PowerAuthAuthenticationException ex) {
-                    // silently ignore here and make sure authentication object is null
+                    // Silently ignore here and make sure authentication object is null
                     request.setAttribute(PowerAuth.AUTHENTICATION_OBJECT, null);
                 }
             }
 
+            // Resolve @PowerAuthEncryption annotation
+            if (powerAuthEncryptionAnnotation != null) {
+                try {
+                    PowerAuthEciesEncryption eciesEncryption = decryptRequest(request);
+                    request.setAttribute(PowerAuthEncryption.ENCRYPTION_OBJECT, eciesEncryption);
+                } catch (PowerAuthEncryptionException ex) {
+                    // Silently ignore here and make sure encryption object is null
+                    request.setAttribute(PowerAuthEncryption.ENCRYPTION_OBJECT, null);
+                }
+            }
         }
 
         return super.preHandle(request, response, handler);
     }
 
+    /**
+     * Decrypt HTTP request body and construct object with ECIES data.
+     *
+     * @param request HTTP request.
+     * @return Object with ECIES data.
+     * @throws PowerAuthEncryptionException In case request decryption fails.
+     */
+    private PowerAuthEciesEncryption decryptRequest(HttpServletRequest request) throws PowerAuthEncryptionException {
+        // Read ECIES metadata from HTTP header
+        final PowerAuthEciesEncryption eciesEncryption = this.encryptionProvider.prepareEciesEncryption(request.getHeader(PowerAuthEncryptionHttpHeader.HEADER_NAME));
+
+        try {
+            // Extract request body from HTTP request
+            Scanner scanner = new Scanner(request.getInputStream()).useDelimiter("\\A");
```

Comment:
I believe you should replace the input stream by resettable stream here. By reading the stream in the scanner, the stream is already read and therefore will not be available for other routines that may need to use it (such as request / response logging, for example).

Reply:
Good catch, I will fix it.

Reply:
Raised in https://github.com/wultra/powerauth-restful-integration/issues/85